### PR TITLE
Decimal v8

### PR DESCRIPTION
**Pull request by losipiuk**
supersedes #4810

### CODE REVIEW

**Comment by martint** [comment]
Re: removing the tests, I actually meant to remove the tests for decimals in partition keys, since it's not supported by all versions of Hive and it's currently hard to specialize for each of the distributions.

Anyway, I just did that and updated this branch: https://github.com/martint/presto/tree/pr4810

File: presto-tpch/src/main/java/com/facebook/presto/tpch/TpchMetadata.java [hunk 1]
```diff
@@ -296,4 +370,29 @@ private long calculateTotalRows(int scaleBase, double scaleFactor)
         }
         return (long) totalRows;
     }
+
+    public static Type getNumericType(TypeSignature numericTypeSignature)
+    {
+        if (numericTypeSignature.getBase().equals(DECIMAL)) {
```
Comment (martint) [inline presto-tpch/src/main/java/com/facebook/presto/tpch/TpchMetadata.java]: I thought there was a discussion about not making the type dynamic for the TPCH connector.

The TPC-H connector should return the expected types according to what the TPC-H spec says. If we want to generate datasets with other types for the purpose of benchmarking or testing, the loading queries should be adjusted to introduce appropriate casts (e.g., in copyTable in the XXXQueryRunners)

Comment (losipiuk) [inline presto-tpch/src/main/java/com/facebook/presto/tpch/TpchMetadata.java]: I was not part of this discussion. But it makes sense to me. 
It seems to me that if we want that approach we can safely drop this commit from PR and change it to desired shape in separate - single commit - PR.

I will try to run the build with this commit dropped and see if it passes.

@sopel39 what do you think.

**Comment by losipiuk** [comment]
> Re: removing the tests, I actually meant to remove the tests for decimals in partition keys, since it's not > supported by all versions of Hive and it's currently hard to specialize for each of the distributions.

> Anyway, I just did that and updated this branch: https://github.com/martint/presto/tree/pr4810


Thanks! Sorry for misunderstatning.

File: presto-tpch/src/main/java/com/facebook/presto/tpch/TpchMetadata.java (hunk 1 above)
Comment (sopel39) [inline presto-tpch/src/main/java/com/facebook/presto/tpch/TpchMetadata.java]: Sure, we can remove remove dynamic decimal support from TPCH connector. We can also remove this commit from this PR and add it later on. Both are fine for me.

Comment (losipiuk) [inline presto-tpch/src/main/java/com/facebook/presto/tpch/TpchMetadata.java]: The build passes without that commit. I vote for removing it and fix it as separate PR so we do not block this one from master integration.

Comment (martint) [inline presto-tpch/src/main/java/com/facebook/presto/tpch/TpchMetadata.java]: Ok, that sounds good.

Comment (martint) [inline presto-tpch/src/main/java/com/facebook/presto/tpch/TpchMetadata.java]: I updated https://github.com/martint/presto/tree/pr4810 to remove that commit. We're cutting a release soon, so I'll merge it as soon as that's done.

**Comment by martint** [comment]
Merged!

**Comment by mattsfuller** [comment]
Nice!!

**Comment by losipiuk** [comment]
Thanks :)

**Pull request closed without merging**